var/www/so/capme/.inc/callback-elastic.php

```diff
@@ -278,17 +278,39 @@ function cliscript($cmd, $pwd) {
 				$errMsgElastic = "Second ES query didn't return a total number of hits.";
 			} elseif ( $elastic_response_object["hits"]["total"] == "0") {
 				$errMsgElastic = "Second ES query couldn't find this ID.";
-			/*} elseif ( $elastic_response_object["hits"]["total"] != "1") {
-				$errMsgElastic = "Second ES query returned multiple matches.";*/
 			} elseif ( ! isset($elastic_response_object["hits"]["hits"][0]["_source"]["protocol"]) ) {
 				$errMsgElastic = "Second ES query didn't return a protocol field.";
 			} elseif ( !in_array($elastic_response_object["hits"]["hits"][0]["_source"]["protocol"], array('tcp','udp'), TRUE)) {
                 		$errMsgElastic = "CapMe currently only supports TCP and UDP.";
-			} else { 
+			} else {
+				
+				// Check to see how many hits we got back from our query
+				$num_records = $elastic_response_object["hits"]["total"];
+				$delta_arr = array();
+
+				// For each hit, we need to compare its timestamp to the timestamp of our original record (from which we pivoted).
+				for ( $i =0 ; $i < $num_records; $i++) {
+                                        $record_ts = $elastic_response_object["hits"]["hits"][$i]["_source"]["timestamp"];
+                                        if ($orig_ts > $record_ts){
+                                                $delta = $orig_ts - $record_ts;
+					}
+                                        elseif ($orig_ts < $record_ts){
+                                                $delta = $record_ts - $orig_ts;
+                                        }
+					else {
+						$delta = 0;
+					}
+                                        $delta_arr[$i] = $delta;
+                                }
+				
+				// Get the key for the hit with the smallest delta
+				$min_val = min($delta_arr);
+				$key = array_search($min_val, $delta_arr);
+
 				// In case we didn't parse out IP addresses and ports above, let's do that now
 				// source_ip
-				if (isset($elastic_response_object["hits"]["hits"][0]["_source"]["source_ip"])) {
-					$sip = $elastic_response_object["hits"]["hits"][0]["_source"]["source_ip"];
+				if (isset($elastic_response_object["hits"]["hits"][$key]["_source"]["source_ip"])) {
+					$sip = $elastic_response_object["hits"]["hits"][$key]["_source"]["source_ip"];
 					error_log($sip);
```

Comment:
Let's go ahead and remove these `error_log()` lines.

Reply:
Will do.